Library/Homebrew/utils/github.rb

```diff
@@ -270,41 +270,77 @@ def upload_release_asset(user, repo, id, local_file: nil, remote_file: nil)
 
   def get_workflow_run(user, repo, pr, workflow_id: "tests.yml", artifact_name: "bottles")
     scopes = CREATE_ISSUE_FORK_OR_PR_SCOPES
-    base_url = "#{API_URL}/repos/#{user}/#{repo}"
-    pr_payload = API.open_rest("#{base_url}/pulls/#{pr}", scopes: scopes)
-    pr_sha = pr_payload["head"]["sha"]
-    pr_branch = URI.encode_www_form_component(pr_payload["head"]["ref"])
-    parameters = "event=pull_request&branch=#{pr_branch}"
-
-    workflow = API.open_rest("#{base_url}/actions/workflows/#{workflow_id}/runs?#{parameters}", scopes: scopes)
-    workflow_run = workflow["workflow_runs"].select do |run|
-      run["head_sha"] == pr_sha
+
+    # GraphQL unfortunately has no way to get the workflow yml name, so we need an extra REST call.
+    workflow_api_url = "#{API_URL}/repos/#{user}/#{repo}/actions/workflows/#{workflow_id}"
+    workflow_payload = API.open_rest(workflow_api_url, scopes: scopes)
+    workflow_id_num = workflow_payload["id"]
+
+    query = <<~EOS
```

Comment:
Would be nice to have a convenience wrapper for constructing GraphQL queries.

Reply:
I'm not sure what that would look like. It's effectively it's own language.

Reply:
Yes, neither do I. I just know it's a pain to write them down. 🙂